### PR TITLE
[php-symfony] Allow Symfony 7 / Remove EOL PHP / Bearer Fix

### PR DESCRIPTION
**Pull request by dmetzner**
- add Symfony 7 support
  - remove support php < 8.2 (EOL)
  - remove symfony < 6.4 support

Bug Fix
  - add missing $security{{name}} variable when using Bearer Auth (https://github.com/Catrobat/Catroweb-API/pull/143/files#r1561127625)

Misc
 - The getContentType method is deprecated; use getContentTypeFormat instead. (Since: symfony/http-foundation', '6.2' deprecated // https://github.com/symfony/symfony/blob/7.1/UPGRADE-7.0.md)
 - use match instead of a switch for simple assignments
 - remove default depth param from json_encode call (`function json_encode(mixed $value, int $flags = 0, int $depth = 512): string|false {}`)
 - make data provider static  - PHPUnit (https://github.com/sebastianbergmann/phpunit/commit/9caafe2d49b33a21f87db248a8ad6ca7c7bdac09)

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/php-symfony/Controller.mustache [hunk 1]
```diff
@@ -230,7 +230,7 @@ class Controller extends AbstractController
                     // add custom format to request
                     $format = $mimeType;
                     $request->setFormat($format, $format);
-                    $currentFormat = $request->getContentType();
+                    $currentFormat = $request->getContentTypeFormat();
```
Comment (dmetzner) [inline modules/openapi-generator/src/main/resources/php-symfony/Controller.mustache:233]: symfony/http-foundation', '6.2', 
'The `getContentType` method is deprecated, use `getContentTypeFormat` instead.'

**Comment by dmetzner** [comment]
@ksm2 @BenjaminHae

@jebentier (2017/07), @dkarlovi (2017/07), @mandrean (2017/08), @jfastnacht (2017/09), [@ybelenko](https://github.com/ybelenko) (2018/07), @renepardon (2018/12)

**Comment by wing328** [comment]
thanks for the PR

i notice that this PR targets 8.0.x (upcoming major release)

is it correct to say that this PR contains breaking changes (without fallbacks)?

**Comment by dkarlovi** [comment]
Removing support for deps is not a breaking change. The user will just not get the update.

**Comment by dmetzner** [comment]
@dkarlovi Thanks for the information. Changed the target branch back to master

**Comment by wing328** [comment]
also tested locally and the result seems good:
```
Tests: 63, Assertions: 22, Deprecations: 116, Skipped: 47, Risky: 20.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:30 min
[INFO] Finished at: 2024-04-13T10:43:13+08:00
[INFO] ------------------------------------------------------------------------
[INFO] 3 goals, 3 executed
[INFO] Cleaning up the local build cache...
[INFO] Deleted 0 unused files from the local build cache in 0.004s
[INFO] ------------------------------------------------------------------------
```

**Pull request merged**